app/scripts/controllers/projectBrowseCatalog.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('ProjectBrowseCatalogController',
+    function($scope,
+             $location,
+             $routeParams,
+             AuthorizationService,
+             AuthService,
+             CatalogService) {
+      AuthService.withUser().then(function() {
```

Comment:
We should really be calling `ProjectsService.get()` to make sure the project exists. In that case, you don't need `AuthService.withUser()`.